lib/core/src/persist/mod.rs

```diff
@@ -224,3 +224,87 @@ impl Persister {
         Ok(payments)
     }
 }
+
+#[cfg(test)]
+pub(crate) mod test_utils {
+    use anyhow::{anyhow, Result};
+    use bip39::rand::{self, distributions::Alphanumeric, Rng};
+    use tempdir::TempDir;
+
+    use super::{Network, PaymentTxData, PaymentType, Persister};
+
+    pub(crate) struct TempPersister {
+        pub(crate) persister: Persister,
+        #[allow(dead_code)]
+        pub(crate) temp_dir: TempDir,
+    }
+
+    pub(crate) fn new_persister() -> Result<TempPersister> {
+        let temp_dir = TempDir::new("liquid-sdk")?;
+        let persister = Persister::new(
```

Comment:
Can we use in memory persister for testing?

Reply:
Unfortunately I don't think this is possible, since we can't override the Persister's `get_connection` method (which is used internally by all other methods).

Reply:
Why not? We can save a flag whether the persister is in memory or not (inside the persister) and change the get_connection accordingly. Or you see an issue with that?

Reply:
I see, didn't think about making it a struct flag, rather an argument to `get_connection`. Will do!